content/zh/blog/2021/external-locality-failover/index.md

```diff
@@ -0,0 +1,242 @@
+---
+title: "为外部服务配置故障转移"
+description: 了解如何为网格外部的端点配置本地负载均衡和故障转移。
+publishdate: 2021-06-04
+attribution: "Ram Vennam (Solo.io)"
+keywords: [locality,region,failover,Istio,outlier,external]
+---
+
+Istio 强大的 API 可用于解决各种使用服务网格时遇到的问题。
+许多用户对其强大的入口和东西向流量能力都比较了解，但是除此之外它也为出口（向外）流量提供了诸多功能。
+当您的应用程序需要与外部服务（例如云服务商提供的数据库端点）通信时，
+该功能就可以派上用场。根据工作负载运行的位置，通常可以有多个端点供用户选择。
+例如，Amazon 的 DynamoDB 就可以提供跨区域的[多个端点](https://docs.aws.amazon.com/general/latest/gr/ddb.html)。
+由于延迟原因，常常希望选择使用离您工作负载最近的端点，
+但是为了防止未按照预期正常工作的情况发生，您可能需要配置自动故障转移使流量可以指向另一个端点。
+
+与在服务网格内运行的服务类似，您可以对 Istio
+进行异常检测配置来实现将流量指向健康端点的故障转移操作，
+并且该配置仍然对您的应用程序完全透明。在此示例中，我们将使用
+Amazon DynamoDB 端点选择与 Google Kubernetes
+Engine（GKE）集群中运行的工作负载处于相同或接近的主要区域。
+另外我们还会为其配置一个故障转移区域。
+
+|路由|端点|
+|--- |--- |
+|主路由|http://dynamodb.us-east-1.amazonaws.com|
+|故障路由|http://dynamodb.us-west-1.amazonaws.com|
+
+![故障转移](./external-locality-failover.png)
+
+## 使用 ServiceEntry 定义外部端点 {#define-external-endpoints-using-a-serviceentry}
+
+[本地负载均衡](/zh/docs/tasks/traffic-management/locality-load-balancing/)基于 `region`
+或 `zone` 工作，这通常是从 Kubernetes 节点上设置的标签推断出来的。
+首先，确定您工作负载所处的位置：
+
+{{< text bash >}}
+$ kubectl describe node | grep failure-domain.beta.kubernetes.io/region
+                    failure-domain.beta.kubernetes.io/region=us-east1
+                    failure-domain.beta.kubernetes.io/region=us-east1
+{{< /text >}}
+
+在此示例中，GKE 集群节点在 `us-east1` 区域中运行。
+
+接下来，创建一个 `ServiceEntry` 来聚合您要使用的端点。在此示例中，
+我们选择了 `mydb.com` 作为 host。这也是您应用程序需要配置的连接地址。
+然后将主要端点的 `locality` 设置为与您工作负载相同的区域：
+
+{{< text yaml >}}
+apiVersion: networking.istio.io/v1beta1
+kind: ServiceEntry
+metadata:
+name: external-svc-dns
+spec:
+hosts:
+- mydb.com
+location: MESH_EXTERNAL
+ports:
+- number: 80
+  name: http
+  protocol: HTTP
+resolution: DNS
+endpoints:
+- address: dynamodb.us-east-1.amazonaws.com
+  locality: us-east1
+  ports:
+    http: 80
+- address: dynamodb.us-west-1.amazonaws.com
+  locality: us-west
+  ports:
+    http: 80
+{{< /text >}}
```

Comment:
The indentations are incorrect, and inconsistent with en upstream now. What's happened?